packages/eslint-plugin/src/rules/no-var-requires.ts

```diff
@@ -43,16 +43,24 @@ export default createRule<Options, MessageIds>({
     function isImportPathAllowed(importPath: string): boolean {
       return allowPatterns.some(pattern => importPath.match(pattern));
     }
+
+    function isStringOrTemplateLiteral(node: TSESTree.Node): boolean {
+      return (
+        (node.type === AST_NODE_TYPES.Literal &&
+          typeof node.value === 'string') ||
+        node.type === AST_NODE_TYPES.TemplateLiteral
+      );
+    }
+
     return {
       'CallExpression[callee.name="require"]'(
         node: TSESTree.CallExpression,
       ): void {
-        if (
-          node.arguments[0]?.type === AST_NODE_TYPES.Literal &&
-          typeof node.arguments[0].value === 'string' &&
-          isImportPathAllowed(node.arguments[0].value)
-        ) {
-          return;
+        if (node.arguments[0] && isStringOrTemplateLiteral(node.arguments[0])) {
+          const argValue = getStaticStringValue(node.arguments[0]);
```

Comment:
[non-actionable] `getStaticStringValue` internally checks for `node.type === AST_NODE_TYPES.Literal || node.type === AST_NODE_TYPES.TemplateLiteral`, so calling `isStringOrTemplateLiteral` seems a bit redundant here.. Although it returns modified values for non-string literals, but in this particular case (with the `require` call) I think it is fine, since passing non-string values to the `require` call would result in a TS error.

I'm fine with the current implementation, just leaving my thoughts here

Reply:
Hi @auvred Thanks for sharing your thoughts.
you're right, I added that condition because I was concerned about cases where non-strings(regex, number,..) are used in the arguments of the `require()` (probably no one writes code like that).
I thought it would be nice if TypeScript Eslint didn't throw an error when non-string value is used and a TS error occurs.

Reply:
> didn't throw an error when (invalid thing)

Aha, yes, that's a good thing to look out for in general. This is a very good + timely thread! 😄 Thanks - #8549